frontend/src/pages/Overview.js

```diff
@@ -19,32 +19,42 @@ import { useNavigate } from "react-router-dom";
 import { copyToClipboard } from "../utils/clipboard";
 import { getTokenPrice } from "../features/common/commonSlice";
 
-export default function Overview() {
+export default function Overview(props) {
+  const { chainID, chainName } = props;
   const wallet = useSelector((state) => state.wallet);
-  const { chainInfo, connected, address } = wallet;
-  const balance = useSelector((state) => state.bank.balance);
-  const delegations = useSelector((state) => state.staking.delegations);
-  const rewards = useSelector((state) => state.distribution.delegatorRewards);
-  const unbonding = useSelector((state) => state.staking.unbonding);
+  const { connected } = wallet;
+  const address = wallet.networks[chainID].walletInfo.bech32Address;
+  const chainInfo = wallet.networks[chainID].network;
+  const pubkey = wallet.networks[chainID].walletInfo.pubKey;
+  const balance = useSelector((state) => state.bank.balances);
+  const delegations = useSelector(
+    (state) => state.staking.chains[chainID].delegations
+  );
+  const rewards = useSelector(
+    (state) => state.distribution.chains[chainID].delegatorRewards
+  );
+  const unbonding = useSelector(
+    (state) => state.staking.chains[chainID].unbonding
+  );
   const account = useSelector((state) => state.auth.accountInfo);
   const priceInfo = useSelector((state) => state.common.tokensInfoState.info);
-
   const selectedAuthz = useSelector((state) => state.authz.selected);
-
+  const { config } = chainInfo;
+  const coinDecimals = config?.currencies[0].coinDecimals || 1;
+  const coinDenom = config?.currencies[0].coinMinimalDenom || "";
+  const dispatch = useDispatch();
   const [available, setTotalBalance] = useState(0);
   const [delegated, setTotalDelegations] = useState(0);
   const [pendingRewards, setTotalRewards] = useState(0);
   const [unbondingDel, setTotalUnbonding] = useState(0);
 
-  const dispatch = useDispatch();
-
-  const { config } = chainInfo;
-  const coinDecimals = config?.currencies[0].coinDecimals || 1;
-  const coinDenom = config?.currencies[0].coinMinimalDenom || "";
-
   useEffect(() => {
     if (connected && config.currencies.length > 0) {
-      setTotalBalance(parseBalance([balance.balance], coinDecimals, coinDenom));
+      if (balance?.[chainID]?.list?.[0] !== undefined)
```

Comment:
The main denomination can be located anywhere in the array. To obtain the balance of the main denomination, you can either use the "balance by denom" API or iterate over the balances array to find the balance associated with the main denomination.

Reply:
done